versioncontrol/peer.go

```diff
@@ -229,22 +232,25 @@ func (versions Versions) ValidateRollouts(log *zap.Logger) error {
 			log.Warn("non-binary field in versions config struct", zap.String("field name", value.Type().Field(i).Name))
 			continue
 		}
-		if err := binary.Rollout.Validate(value.Type().Field(i).Name, log); err != nil {
+		if err := binary.Rollout.Validate(); err != nil {
+			if err == emptySeedErr {
+				log.Warn(err.Error(), zap.String("binary", value.Type().Field(i).Name))
+				continue
+			}
 			validationErrs.Add(err)
 		}
 	}
 	return validationErrs.Err()
 }
 
 // Validate validates the rollout seed and cursor config values.
-func (rollout Rollout) Validate(binary string, log *zap.Logger) error {
+func (rollout Rollout) Validate() error {
 	seedLen := len(rollout.Seed)
 	if seedLen == 0 {
-		log.Warn("empty seed", zap.String("binary", binary))
-		return nil
+		return emptySeedErr
```

Comment:
:thinking: a caller from outside of this package may be interested in knowing about this error.

I see that `Rollout` type isn't currently used by any other package except by a test (package `version_test`) so we could export it in the future when needed, however, we are already exporting `RolloutErr` and classifying all the errors returned by this function with it.

I think that the caller should be able to distinguish the `emptySeedErr` just comparing with it (.i.e `==`) rather than discriminating not being from the `RolloutErr` class (i.e. `!RolloutErr.Has(err)`).

If you export it, then, add to the doc function comment about this behavior.